rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -425,13 +425,27 @@ void game_list_frame::Refresh(const bool from_drive, const std::vector<std::stri
 			m_progress_dialog->SetValue(0);
 		}
 
-		// Remove the specified serials (title id) in "games.yml" file (if any)
-		Emu.RemoveGames(serials_to_remove_from_yml);
-
 		const std::string games_dir = rpcs3::utils::get_games_dir();
-		const u32 games_added = Emu.AddGamesFromDir(games_dir);
 
-		if (games_added)
+		// List of serials (title id) to remove in "games.yml" file (if any)
+		std::vector<std::string> serials_to_remove = serials_to_remove_from_yml; // Initialize the list with the specified serials (if any)
+
+		// Scan game list to detect the titles belonging to auto-detection "games" folder
+		for (const auto& [serial, path] : Emu.GetGamesConfig().get_games()) // Loop on game list file
+		{
+			// NOTE: Used starts_with(games_dir) instead of Emu.IsPathInsideDir(path, games_dir) due the latter would check also the existence of the paths
+			//
+			if (path.starts_with(games_dir)) // If game path belongs to auto-detection "games" folder, add the serial to the removal list
+			{
+				serials_to_remove.push_back(serial);
+			}
+		}
```

Comment:
why is this done here unconditionally when you literally do the same loop in the function that cleans the list?

Reply:
because `games` is an auto-detection folder meant just to dynamically update the game list (without the need to manually add or remove them; e.g. by the use of `File->Add games`). You move game in that folder, then you delete some games, add new game etc. and the game list in `games.xml` is also expected to be consistent with the content present in the `games` folder.
The Cleanup function in the `All Title` menu is meant to provide an overall cleanup for lost/corrupted games (e.g. added by the use of `File->Add games`). The target of that function is global.